### PR TITLE
Add hasPublicItems property to commons #2049

### LINKED ISSUE #2049: Add hasPublicItems property to commons
Keep track of whether a common has any public items in the feed or not
could be updated upon creation of new streams and possibly upon deletion of a public item

### PR DESCRIPTION
**Pull request by ariel-dao**
Add tracker for common on public feed items based on filtering out pinned finned items and having an empty circle visibility - runs on update/delete and custom logic for add feed item

And a migration

### CODE REVIEW

File: functions/src/commons/business/checkForPublicItems.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  return !(
+    await db
+      .collectionGroup(SUBCOLLECTIONS.CommonFeed)
+      .where("isDeleted", "==", false)
+      .where("commonId", "==", commonId)
+      .where("circleVisibility", "==", [])
+      .where("id", "not-in", commonPinnedItemsIds)
+      .get()
+  ).empty;
```
Comment (danielr18) [inline functions/src/commons/business/checkForPublicItems.ts]: Will this require an index?

File: functions/src/commons/controllers/getFeedItems.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export const getFeedItems = async (
         commonId: string().uuid().required(),
         startAfter: string().optional(),
         limit: number().optional(),
-        userId: string(),
+        userId: string().optional(),
         includeSubCommonsFeedItems: boolean().default(false),
       }).validate({
         userId: req.user.uid,
```
Comment (danielr18) [inline functions/src/commons/controllers/getFeedItems.ts]: Should it become `req.user?.uid`?

**Review by danielr18 — CHANGES_REQUESTED** [review]

File: functions/src/commons/business/checkForPublicItems.ts (hunk 1 above)
Comment (ariel-dao) [reply functions/src/commons/business/checkForPublicItems.ts]: added

**Comment by ariel-dao** [comment]
@danielr18 @elatif2020 pinned items (general discussions) are filtered when checking for public items, should I change it?

aside from that, can I merge this?

**Comment by elatif2020** [comment]
> @danielr18 @elatif2020 pinned items (general discussions) are filtered when checking for public items, should I change it?
> 
> aside from that, can I merge this?

Why are we filtering out pinned items? @ariel-dao @andreymikhadyuk

**Comment by ariel-dao** [comment]
- [x] Remove filtering by pinned/general

**Review by danielr18 — APPROVED** [review]

**Pull request merged**